File: src/main/java/com/prova/swapi/model/Planeta.java
package com.prova.swapi.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

/**
 * Representa um planeta.
 */
@Data
public class Planeta {
    private String name;
    private String population;
    private String climate;
    @JsonProperty("rotation_period")
    private String rotationPeriod;
    @JsonProperty("orbital_period")
    private String orbitalPeriod;
    private String diameter;
    private String gravity;
    private String terrain;
    @JsonProperty("surface_water")
    private String surfaceWater;
}

File: swfrontend/src/app/components/planetas/models/planeta.ts
export class Planeta {

    nome: string;
    populacao: number;
    clima: string;
    periodoRotacao: string;
    peridoOrbital: string;
    diametro: string;
    gravidade: string;
    terreno: string;
    qtdAguaSuperficie: string;

    constructor(item) {
        this.nome = item.name;
        this.populacao = item.population;
        this.clima = item.climate;
        this.periodoRotacao = item.rotation_period;
        this.peridoOrbital = item.orbital_period;
        this.diametro = item.diameter;
        this.gravidade = item.gravity;
        this.terreno = item.terrain;
        this.qtdAguaSuperficie = item.surface_water;
    }
}
